input/otlp/metrics.go

```diff
@@ -143,47 +167,53 @@ func (c *Consumer) addMetric(metric pmetric.Metric, ms metricsets) bool {
 			if sample, ok := numberSample(dp, modelpb.MetricType_METRIC_TYPE_GAUGE); ok {
 				sample.Name = metric.Name()
 				ms.upsert(dp.Timestamp().AsTime(), dp.Attributes(), sample)
+				*remainingDataPoints--
 			} else {
 				anyDropped = true
 			}
 		}
-		return !anyDropped
 	case pmetric.MetricTypeSum:
 		dps := metric.Sum().DataPoints()
 		for i := 0; i < dps.Len(); i++ {
 			dp := dps.At(i)
 			if sample, ok := numberSample(dp, modelpb.MetricType_METRIC_TYPE_COUNTER); ok {
 				sample.Name = metric.Name()
 				ms.upsert(dp.Timestamp().AsTime(), dp.Attributes(), sample)
+				*remainingDataPoints--
 			} else {
 				anyDropped = true
 			}
 		}
-		return !anyDropped
 	case pmetric.MetricTypeHistogram:
 		dps := metric.Histogram().DataPoints()
 		for i := 0; i < dps.Len(); i++ {
 			dp := dps.At(i)
 			if sample, ok := histogramSample(dp.BucketCounts(), dp.ExplicitBounds()); ok {
 				sample.Name = metric.Name()
 				ms.upsert(dp.Timestamp().AsTime(), dp.Attributes(), sample)
+				*remainingDataPoints--
 			} else {
 				anyDropped = true
```

Comment:
Should we be removing this line (and similar lines)? We're not dropping the whole metric, just the data point.